### ISSUE TITLE
Change OpenEye and RDKit SDF reading functionality to read all molecules in a file 

### ISSUE THREAD

**Issue by hmacdope** [choderalab/asapdiscovery#949]
In @apayne97 great work in #692 I have been thinking (dangerous) and think that only returning the **first** molecule by default in an SDF file is probably not the API design we want. 

This is a legacy of way-way back pre 0.1.0. Instead we should return all the molecules in the file and it should be on the downstream consumer of these to parse the relevant molecules or decide what to do.  

This will have to wait until 0.4.0+ as we have other targets we need to hit first and will require quite a bit of refactoring.

**Comment by hmacdope** [comment]
Would be curious to hear your thoughts @apayne97 @kaminow

**Comment by kaminow** [comment]
yeah I think I agree with this. for me at least, the reasoning behind returning one molecule by default was based on not breaking everything else that was built on the `Ligand` object. but if we're willing to do that and rewrite some stuff, then yeah let's do it, I think that behavior makes more sense ultimately

**Comment by apayne97** [comment]
Yeah I agree we should support that. I've also been trying to avoid breaking things. I think the best thing would be to use the MolFileFactory to guide how an sdf file is imported and make the openeye and rdkit functions read everything in the file.
